sklearn/utils/estimator_checks.py

```diff
@@ -127,8 +128,9 @@ def _yield_regressor_checks(name, Regressor):
     if name != 'CCA':
         # check that the regressor handles int input
         yield check_regressors_int
-    # Test if NotFittedError is raised
-    yield check_estimators_unfitted
+    if name != "GaussianProcessRegressor":
```

Comment:
GaussianProcessRegressor can predict without being fitted by using its prior (essentially specified by the kernel). This is useful for checking which kind of assumptions are made by the different kernels, see for instance the example plot_gpr_prior_posterior.py for various kernels.